grails-app/taglib/org/pih/warehouse/SelectTagLib.groovy

```diff
@@ -823,19 +795,10 @@ class SelectTagLib {
 
     def renderNoSelectionOption = { noSelectionKey, noSelectionValue, value ->
         // If a label for the '--Please choose--' first item is supplied, write it out
-        out << '<option value="' << (noSelectionKey == null ? "" : noSelectionKey) << '"'
-        if (noSelectionKey.equals(value)) {
-            out << ' selected="selected" '
+        out << "<option value=\"${noSelectionKey == null ? '' : noSelectionKey}\""
+        if (noSelectionKey == value) {
+            out << ' selected="selected"'
         }
-        out << '>' << noSelectionValue.encodeAsHTML() << '</option>'
+        out << ">${noSelectionValue.encodeAsHTML()}</option>"
     }
-
-    private String optionValueToString(def el, def optionValue) {
-        if (optionValue instanceof Closure) {
-            return optionValue(el).toString().encodeAsHTML()
-        }
-
-        el[optionValue].toString().encodeAsHTML()
-    }
-
```

Comment:
OK, if you've made it this far, you've read all the taglib code. Taglibs don't listen to `grails.views.default.codec` and need `encodeAsHTML()` sprinkled throughout. GSP pages, on the other hand, do follow this config setting, and (almost) every `encodeAsHTML()` call within them is incorrect.

This stuff changes (for the better) in Grails 3.